### PR TITLE
Improve Reflectometry Help Facilities

### PR DESCRIPTION
**Pull request by edwardb96**
- Adds the usual `?` button to the reflectometry interface which links to it's corresponding documentation.
- Makes line edit hints and tooltips readable on Linux distros.
- Sets tooltips to corresponding property documentation on the Reflectometry settings GUI.
- Adds more tooltips to the Runs and Event tabs, excluding the `Save ASCII` tab for now due to imminent changes.

**To test:**
1. Open the reflectometry GUI and ensure that the `?` button in the lower left corner opens the help browser as expected.
2. On Linux 
   1. Go to the settings tab and click in the `Stitch1DMany` box, begin typing and ensure that the suggestions are readable.
   2. Returning to the runs tab, press the `->?` button on the toolbox to enable 'What's this' mode.
   3. Click on one of the other tools in the toolbox and ensure that the help is also readable.
3. Load the sample table `INTER_NR_test2.tbl` from the ISIS sample data by clicking `Reflectometry->Import TBL` and selecting the file. Process all the runs before switching to the settings tab and editing a setting. Ensure that the rows are no longer highlighted in green.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by josephframsay — DISMISSED** [review]
Changes look mostly good, but unfortunately the runs are still highlights in a rather attractive green on Linux.

**Review by josephframsay — CHANGES_REQUESTED** [review]
Changes look mostly good, but unfortunately the rows are still highlighted in green on both Linux and Windows (see screenshot)

screenshot from 2018-02-12 12-14-56

File: MantidPlot/src/ApplicationWindow.cpp [hunk 1]
```diff
@@ -11558,6 +11558,34 @@ void ApplicationWindow::setUpdateCurvesFromTable(Table *table, bool on) {
...
+  The default pallete shows light text on a pale background which, although not
+  affecting tooltips, makes LineEdit hints and 'What's This' boxes difficuilt
+  if not impossible to read.
+
+  Changing the tooltip text to a darker colour fixes the problem for 'LineEdit'
+  hints and 'What's This' boxes but creates one for ordinary tooltips.
+  Instead we set the background colour to a darker one, consistent with
+  ordinary tooltips.
+
+  @returns A pallete with contrasting tooltip and background colours.
```
[the remark this answers is not in the document]
Comment (edwardb96) [reply MantidPlot/src/ApplicationWindow.cpp]: Good catch

**Review by josephframsay — APPROVED** [review]

**Comment by josephframsay** [comment]
My bad, misread the instructions. Tests work as expected - all good to go!

**Pull request merged**